1.14/test/run

```diff
@@ -339,65 +350,68 @@ test_perl_directive() {
   test_for_output "http://$(container_ip):${test_port}/" "X-Perl-Version: ${perl_version}"
-  check_result $? "perl-test-app"
+  check_result $?
 
   info "Testing perl directive runs perl location handler"
   test_for_output "http://$(container_ip):${test_port}/perl" "Perl location handler is working"
-  check_result $? "perl-test-app"
+  check_result $?
 
   cleanup_test_app
 }
 
-cid_file=$(mktemp -u --suffix=.cid)
-
-# Since we built the candidate image locally, we don't want S2I attempt to pull
-# it from Docker hub
-s2i_args="--pull-policy=never"
-
-prepare "test-app"
-run_s2i_build "test-app"
-check_result $? "test-app"
+test_application_user() {
+  # Test application with random uid
+  CONTAINER_ARGS="--user 12345" test_application
+}
 
-# Verify the 'usage' script is working properly when running the base image with 's2i usage ...'
-test_s2i_usage
-check_result $? "test-app"
 
-# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
-test_docker_run_usage
-check_result $? "test-app"
+function run_all_tests() {
+  for test_case in $TEST_LIST; do
+    info "Running test $test_case ...."
+    if $test_case; then
+      printf -v test_short_summary "${test_short_summary}[PASSED] $test_case\n"
+    else
+      printf -v test_short_summary "${test_short_summary}[FAILED] $test_case\n"
+      [ -n "${FAIL_QUICKLY:-}" ] && echo "$sum" && return 1
+    fi
+  done;
+  cleanup "${APP_NAME}"
+}
 
-# Test application with default uid
-test_application
+function build_image() {
+  local test_name="$1"
+  cid_file=$(mktemp -u --suffix=.cid)
 
-# Test logging works
-test_log_output
+  # Since we built the candidate image locally, we don't want S2I attempt to pull
+  # it from Docker hub
+  s2i_args="--pull-policy=never"
 
-# Test logging to volume works
-test_log_volume_output
+  prepare "${test_name}"
+  run_s2i_build "${test_name}"
+  check_result $? "${test_name}"
+}
 
-# Test application with random uid
-CONTAINER_ARGS="--user 12345" test_application
-cleanup "test-app"
+build_image "test-app"
 
-# Test start-hook script functionality
-cid_file=$(mktemp -u --suffix=.cid)
+# Run the chosen tests
+TEST_LIST=${TEST_LIST_APP} run_all_tests
 
-prepare "start-hook-test-app"
-run_s2i_build "start-hook-test-app"
-check_result $? "start-hook-test-app"
+build_image "start-hook-test-app"
 
-test_pre_init_script
-cleanup "start-hook-test-app"
+# Run the chosen tests
+TEST_LIST=${TEST_LIST_HOOK_APP} run_all_tests
 
-# Test perl module functionality
-cid_file=$(mktemp -u --suffix=.cid)
+build_image "perl-test-app"
 
-prepare "perl-test-app"
-run_s2i_build "perl-test-app"
-check_result $? "perl-test-app"
+# Run the chosen tests
+TEST_LIST=${TEST_LIST_PERL_APP} run_all_tests
 
-test_perl_directive
+echo "$test_short_summary"
 
-cleanup "perl-test-app"
+if [ $TESTSUITE_RESULT -eq 0 ] ; then
+  echo "Tests for ${IMAGE_NAME} succeeded."
+else
+  echo "Tests for ${IMAGE_NAME} failed."
```

Comment:
`return $TESTSUITE_RESULT` here?

Reply:
It has to be exit and not return. Return if from function ;)

Reply:
Fixed by https://github.com/sclorg/nginx-container/pull/117/commits/4ca4b3b3d45b7b378c289efbf50aa37a8ff5abc2